### ISSUE TITLE
Missing type annotation in declaration of variable binding

### ISSUE THREAD

**Issue by stanciuadrian** [rust-analyzer/rust-analyzer#4322]
When variable bindings are split in 2 (declaration + initialization, as described in https://doc.rust-lang.org/stable/rust-by-example/variable_bindings/declare.html), the declaration is missing the type annotation. The `x` tooltip is indicating `{unknown}`. `y` has been added to show that ra is properly installed and working.

```rust
fn main() {
    let x;
    x = 3;
    let y = 5;
    println!("{} {}", x, y);
}
```
Looks like this:

<img width="235" alt="radecl" src="https://user-images.githubusercontent.com/2334569/81090230-19951d80-8f06-11ea-8f45-38818e9de928.png">

ra v0.2.159 from VSCode v1.44.2 marketplace on W64

**Comment by matklad** [comment]
I am actually surprised that we don't handle this. I think the fix should be applied somewhere here: https://github.com/rust-analyzer/rust-analyzer/blob/58f64313fdbb215b29cceb843e06d2c1541d9f56/crates/ra_hir_ty/src/infer/expr.rs#L530. I would guess that, instead `TyUnknown`, we should use type var. cc @flodiebold

**Comment by lnicola** [comment]
Fixed in #4412? CC @kiljacken

**Comment by kiljacken** [comment]
Presumably
